packages/rooks/src/hooks/useResizeObserverRef.ts

```diff
@@ -20,26 +19,35 @@ const config: ResizeObserverOptions = {
  * @see https://rooks.vercel.app/docs/useResizeObserverRef
  */
 function useResizeObserverRef(
-  callback: ResizeObserverCallback,
+  callback: ResizeObserverCallback | undefined,
   options: ResizeObserverOptions = config
 ): [CallbackRef] {
+  const { box } = options;
   const [node, setNode] = useState<HTMLElementOrNull>(null);
-  const freshCallback = useFreshTick(callback);
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
+
+  const handleResizeObserver = useCallback<ResizeObserverCallback>(
+    (...args) => {
+      callbackRef.current?.(...args);
+    },
+    []
+  );
 
   useEffect(() => {
-    if (node) {
+    if (node && callbackRef.current) {
```

Comment:
Looks great. I don't think we need to check for `callbackRef.current` here

```suggestion
    if (node) {
```

Reply:
Oh, that's right! `handleResizeObserver` is registered as a callback, so set a node without a callback then set a callback, the callback may not fire the node is not being observed! Thank you!